pkg/operator/osd/osd.go

```diff
@@ -88,48 +155,92 @@ func (c *Cluster) makeDaemonSet() (*extensions.DaemonSet, error) {
 			Annotations: map[string]string{},
 		},
 		Spec: v1.PodSpec{
-			Containers:    []v1.Container{c.osdContainer()},
+			Containers:    []v1.Container{c.osdContainer(devices, directories, selection, config)},
 			RestartPolicy: v1.RestartPolicyAlways,
-			Volumes: []v1.Volume{
-				{Name: k8sutil.DataDirVolume, VolumeSource: dataDirSource},
-				{Name: "devices", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/dev"}}},
-			},
+			Volumes:       volumes,
 		},
 	}
+}
 
-	ds.Spec = extensions.DaemonSetSpec{Template: podSpec}
+func (c *Cluster) osdContainer(devices []StorageDevice, directories []StorageDirectory, selection StorageSelection, config StorageConfig) v1.Container {
+	var commandBuffer bytes.Buffer
 
-	return ds, nil
-}
+	commandBuffer.WriteString(fmt.Sprintf("/usr/bin/rookd osd --config-dir=%s ", k8sutil.DataDir))
 
-func (c *Cluster) osdContainer() v1.Container {
+	// only 1 of device list, device filter and use all devices can be specified.  We prioritize in that order.
+	var deviceFilterEnvVar string
+	if len(devices) > 0 {
+		deviceNames := make([]string, len(devices))
+		for i := range devices {
+			deviceNames[i] = devices[i].Name
+		}
+		commandBuffer.WriteString(fmt.Sprintf("--data-devices=%s ", strings.Join(deviceNames, ",")))
```

Comment:
yes, i will consider using env vars for all of them and see how that shakes out.